language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionNodeValidator.java

```diff
@@ -0,0 +1,246 @@
+    }
+
+    @Override
+    public Boolean transform(LetExpressionNode node) {
+        return isVisited(node) || !node.letKeyword().isMissing() 
+                && !node.inKeyword().isMissing()
+                && node.letVarDeclarations().stream().allMatch(arg -> arg.apply(this))
+                && node.expression().apply(this);
+    }
+
+    @Override
+    public Boolean transform(LetVariableDeclarationNode node) {
+        return isVisited(node) || !node.equalsToken().isMissing()
+                && node.typedBindingPattern().apply(this)
+                && node.expression().apply(this);
+    }
+
+    @Override
+    public Boolean transform(SpreadFieldNode node) {
+        return isVisited(node) || !node.ellipsis().isMissing()
+                && node.valueExpr().apply(this)
+                && node.parent().apply(this);
+    }
+
+    @Override
+    public Boolean transform(MappingConstructorExpressionNode node) {
+        return isVisited(node) || !node.openBrace().isMissing() 
+                && !node.closeBrace().isMissing()
+                && node.fields().stream().allMatch(arg -> arg.apply(this))
+                && node.parent().apply(this); 
+    }
+
+    @Override
+    public Boolean transform(FieldAccessExpressionNode node) {
+        return isVisited(node) || node.expression().apply(this) 
+                && !node.dotToken().isMissing()
+                && node.fieldName().apply(this)
+                && node.parent().apply(this);
+    }
+    @Override
+    public Boolean transform(PositionalArgumentNode node) {
+        return isVisited(node) || node.expression().apply(this)
+                && node.parent().apply(this);
+    }
+
+    @Override
+    public Boolean transform(BasicLiteralNode node) {
+        return isVisited(node) || !node.literalToken().isMissing()
+            && node.parent().apply(this);
+    }
+
+    @Override
+    public Boolean transform(NamedArgumentNode node) {
+        return isVisited(node) || !node.equalsToken().isMissing()
+                && node.argumentName().apply(this) 
+                && node.expression().apply(this)
+                && node.parent().apply(this);
+    }
+    
+    @Override
+    public Boolean transform(RestArgumentNode node) {
+        return isVisited(node) || !node.ellipsis().isMissing()
+                && node.expression().apply(this)
+                && node.parent().apply(this);
+    }
+
+    @Override
+    public Boolean transform(ListConstructorExpressionNode node) {
+        return isVisited(node) || !node.openBracket().isMissing() && !node.closeBracket().isMissing()
+                && node.expressions().stream().allMatch(arg -> arg.apply(this))
+                && node.parent().apply(this);
+    }
+
+    /**
+     * Checks whether the syntax is valid.
+     *
+     * @param node    Node at cursor position
+     * @return {@link Boolean} True if syntactically correct, false otherwise
+     */
+    public static Boolean validate(NonTerminalNode node) {
```

Comment:
Shall we change the return type here to `boolean` primitive type? You will have to do a null check before returning.

Reply:
Done 👍